### PR TITLE
Filter out version = -1.-1.-1 & tests: SDK Muxer

### PR DESCRIPTION
**Pull request by ghost**
version = -1.-1.-1 is a placeholder version and should not be considered
add tests for the same...

Other:
-test directories are isolated; we don't need to delete SDK test versions from them anymore
-better failure comment for the global.json lookup
-combining methods: 'resolve_sdk_version' and 'resolve_patch_sdk_version' 

Addresses: https://github.com/dotnet/core-setup/issues/1760

@dotnet/dotnet-cli

### CODE REVIEW

**Comment by johnbeisner** [comment]
@steveharter 
@wtgodbe 
Can someone take a look at this PR please.

File: src/corehost/cli/fxr/fx_muxer.cpp [hunk 1]
```diff
@@ -796,7 +796,8 @@ bool fx_muxer_t::resolve_sdk_dotnet_path(const pal::string_t& own_dir, const pal
         }
     }
 
-    if (!cli_version.empty())
+    pal::string_t neg_version = _X("|-1.-1.-1|");
```
Comment (steveharter) [inline src/corehost/cli/fxr/fx_muxer.cpp]: Where does the pipe `|` come from? Special character?

File: src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs [hunk 2]
```diff
@@ -190,6 +190,55 @@ public void SdkLookup_Global_Json_Patch_Rollup()
...
+                .Environment(s_DefaultEnvironment)
+                .EnvironmentVariable("DOTNET_MULTILEVEL_LOOKUP", "0")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdErrContaining(Path.Combine(_exeSelectedMessage, "9999.0.4", _dotnetSdkDllMessageTerminator));
+        }
```
Comment (steveharter) [inline src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: No DeleteAvailableSdkVersions?

Note I suspect tests can collide with each other, we should either make sure they have unique versions (across tests) or run the tests in series. I was planning on investigating this.

File: src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs [hunk 3]
```diff
@@ -279,9 +327,6 @@ public void SdkLookup_Must_Pick_The_Highest_Semantic_Version()
                 .Pass()
                 .And
                 .HaveStdErrContaining(Path.Combine(_exeSelectedMessage, "10000.0.0", _dotnetSdkDllMessageTerminator));
-
-            // Remove dummy folders from user dir
-            DeleteAvailableSdkVersions(_userSdkBaseDir, "9999.0.2");
```
Comment (steveharter) [inline src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: Causing a random issue? I notice there are other sdk versions added for this test as well that are not removed (see above on comment on possible colliding)

File: src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs (hunk 2 above)
Comment (johnbeisner) [inline src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: Test directories are isolated. Each test creates a numbered directory [0,1,2 in this particular case corresponding to the 3 'facts' in ''GivenThatICareAboutMultilevelSDKLookup'] with "cmd", "exe", and "user" as subdirectories - no collisions.

Comment (johnbeisner) [inline src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: So 'DeleteAvailableSdkVersions' might be useful if we wanted to test the de-installation of an SDK version.

Comment (steveharter) [inline src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: OK I randomly get exceptions regarding unable to delete directory...

Comment (steveharter) [inline src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: Tests running simultaneously could get the same ID (numbered directory) in a race condition. Not sure if that is the reason for the random failures I get.

**Review by steveharter — APPROVED** [review]

File: src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs (hunk 2 above)
Comment (johnbeisner) [inline src/test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: "Tests running simultaneously could get the same ID" - if this were the case, the tests are invalid as they would mix the versions from the tests running simultaneously.

**Pull request merged**